sdk/src/types/block/protocol/mod.rs

```diff
@@ -263,6 +263,62 @@ impl ProtocolParameters {
             max: self.max_committable_age(),
         }
     }
+
+    /// Calculates the past bounded slot for the given slot of the SlotCommitment.
+    /// Given any slot index of a commitment input, the result of this function is a slot index
+    /// that is at least equal to the slot of the block in which it was issued, or higher.
+    /// That means no commitment input can be chosen such that the index lies behind the slot index of the block,
+    /// hence the past is bounded.
+    pub fn past_bounded_slot(&self, slot_commitment_id: SlotCommitmentId) -> SlotIndex {
+        slot_commitment_id.past_bounded_slot(self.max_committable_age())
+    }
+
+    /// Calculates the future bounded slot for the given slot of the SlotCommitment.
+    /// Given any slot index of a commitment input, the result of this function is a slot index
+    /// that is at most equal to the slot of the block in which it was issued, or lower.
+    /// That means no commitment input can be chosen such that the index lies ahead of the slot index of the block,
+    /// hence the future is bounded.
+    pub fn future_bounded_slot(&self, slot_commitment_id: SlotCommitmentId) -> SlotIndex {
+        slot_commitment_id.future_bounded_slot(self.min_committable_age())
+    }
+
+    /// Returns the slot at the end of which the validator and delegator registration ends and the voting power
+    /// for the epoch is calculated.
+    pub fn registration_slot(&self, epoch_index: EpochIndex) -> SlotIndex {
+        epoch_index.registration_slot(
+            self.genesis_slot(),
+            self.slots_per_epoch_exponent(),
+            self.epoch_nearing_threshold(),
+        )
+    }
+
+    /// TODO what is a good name + description for these methods?
```

Comment:
Maybe something like `delegation_start/end_epoch` or is this also interesting for something else?

Reply:
Not that I'm aware of